zyte_spider_templates/spiders/ecommerce.py

```diff
@@ -241,8 +256,37 @@ def get_start_request(self, url):
         )
 
     def start_requests(self) -> Iterable[Request]:
-        for url in self.start_urls:
-            yield self.get_start_request(url)
+        if self.args.search_queries:
+            for url in self.start_urls:
+                meta: Dict[str, Any] = {
+                    "crawling_logs": {"page_type": "searchRequestTemplate"},
+                }
+                if self.args.crawl_strategy == EcommerceCrawlStrategy.full:
+                    meta["page_params"] = {"full_domain": get_domain(url)}
+                if self.args.extract_from == ExtractFrom.browserHtml:
+                    meta["zyte_api_provider"] = {"browserHtml": True}
```

Comment:
@wRAR Tried `meta["inject"] = [BrowserResponse]` here but it did not seem to work. Should it work? (as in, do we want it to work as a way to force browser over HTTP for AnyResponse, regardless of whether or not it works like that at the moment)

Reply:
@wRAR @kmike My bad, I tested without also declaring DynamicDeps on the callback, I see that’s also required.